### PR TITLE
bench: fix async_pummel1 to current API, watchers: const deinit ptr

### PR DESCRIPTION
**Pull request by linuxy**

### CODE REVIEW

**Comment by mitchellh** [comment]
Thanks, this looks great to me. THere are probably a lot more places a const ptr makes sense. I am hoping the Zig compiler fixes the performance issues around that soon (as it was in stage1) but in the mean time making them const where it makes sense is fine with me.

**Comment by mitchellh** [comment]
Also, this reminds me that I should add _building_ the bench and full examples dir to CI.

File: src/bench/async_pummel_1.zig [hunk 1]
```diff
@@ -75,6 +76,7 @@ fn asyncCallback(
 }
 
 fn threadMain(loop: *xev.Loop) !void {
-    while (state == .running) try notifier.notify(loop);
+    while (state == .running) try notifier.notify();
     state = .stopped;
+    _ = loop;
 }
```
Comment (mitchellh) [inline src/bench/async_pummel_1.zig:82]: We can just not pass the loop into the thread anymore.

**Comment by mitchellh** [comment]
Thanks! I also just updated the GH action to build all examples and benchmarks 😄

**Comment by linuxy** [comment]
Welcome! It's cool seeing a Nix+Zig CI flow.

**Pull request merged**